crates/oci-distribution/src/client.rs

```diff
@@ -89,6 +89,10 @@ impl Client {
 
         let (manifest, digest) = self.pull_manifest(image).await?;
 
+        if manifest.layers.is_empty() {
```

Comment:
I think this is the place we should check for the layer's media type to be `WASM_LAYER_MEDIA_TYPE`.
(this, or in `pull_layer`)

Reply:
agreed. I was trying to figure out how we could assert this check in a more generic fashion. I was concerned that projects like wasme used a different media type. Some more research suggests I was wrong, so it should be safe to validate against this media type.

https://github.com/solo-io/wasm/blob/68e67db5ccba6b2e67b835b9e53225a992f5462a/tools/wasme/pkg/model/image.go#L44